src/objc/smtp/MCOSMTPLoginOperation.mm

```diff
@@ -56,8 +58,24 @@ - (void)operationCompleted
     if (_completionBlock == NULL)
         return;
     
-    NSError * error = [NSError mco_errorWithErrorCode:MCO_NATIVE_INSTANCE->error()];
-    _completionBlock(error);
+    nativeType *op = MCO_NATIVE_INSTANCE;
```

Comment:
Could you actually create in MCOSMTPOperation+Private.h, a method:

```
- (NSError *) error
{
    NSError * error = [NSError mco_errorWithErrorCode:MCO_NATIVE_INSTANCE->error()];
   if (error != nil) {
        MCOSMTPSession *session = [self session];
        if (nativeOp->lastSMTPResponse() || nativeOp->lastSMTPResponseCode()) {
            NSMutableDictionary * userInfo = [[error userInfo] mutableCopy];
            if (nativeOp->lastSMTPResponse()) {
                userInfo[MCOSMTPResponseKey] = MCO_TO_OBJC(nativeOp->lastSMTPResponse());
            }
            if (nativeOp->lastSMTPResponseCode()) {
                userInfo[MCOSMTPResponseCodeKey] = @(session nativeOp());
            }
            error = [NSError errorWithDomain:[error domain] code:[error code] userInfo:userInfo];
        }
   }
  return error;
}
```

And use that method everywhere that code is duplicated.